normandy/recipes/tests/test_models.py

```diff
@@ -119,3 +120,27 @@ def test_filter_by_country_many(self):
         assert recipe.matches(client1)
         assert recipe.matches(client2)
         assert not recipe.matches(client3)
+
+    def test_filter_by_sample_rate(self):
+        always_match = RecipeFactory(sample_rate=100)
+        never_match = RecipeFactory(sample_rate=0)
+        client = ClientFactory()
+
+        assert always_match.matches(client)
+        assert not never_match.matches(client)
+
+    def test_filter_exclude(self):
+        recipe = RecipeFactory(enabled=False)
+        client = ClientFactory()
+
+        assert not recipe.matches(client)
+        assert recipe.matches(client, exclude=[match_enabled])
+
+    def test_filter_exclude_many(self):
```

Comment:
It's meant to test the exclusion on a multivalue matcher field, but doesn't actually use multiple values. I'll write a better name or make it actually test multiple values.